### PR TITLE
Return value of BackedEnum when reading Doctrine entity property.

### PR DESCRIPTION
**Pull request by juniwalk**
See #1080

### CODE REVIEW

**Comment by juniwalk** [comment]
@radimvaculik Alright, this should be much better, I closed #1080.

**Comment by juniwalk** [comment]
@radimvaculik ping

**Comment by radimvaculik** [comment]
@juniwalk How does it work on PHP 8.0, where BackedEnum is not implemented?

**Comment by juniwalk** [comment]
@radimvaculik I think class is not autoloaded so it just ignores it. That is why php8.0 did not fail.

File: src/Row.php [hunk 1]
```diff
@@ -216,6 +217,10 @@ public function getDoctrineEntityProperty(mixed $item, mixed $key): mixed
 			return (string) $value;
 		}
 
+		if (is_object($value) && $value instanceof BackedEnum) {
```
Comment (radimvaculik) [inline src/Row.php:220]:
```suggestion
		if (interface_exists(\BackedEnum::class) && $value instanceof BackedEnum) {
```

Comment (radimvaculik) [reply src/Row.php:220]: @juniwalk I prefer something like this.

Comment (juniwalk) [reply src/Row.php:220]: No problem, I will add it.

Comment (radimvaculik) [reply src/Row.php:220]: Why do you check it, if `$value` is object with `is_object` ?

Comment (juniwalk) [reply src/Row.php:220]: Oh I wish I knew :D I usually don't do that.

**Comment by juniwalk** [comment]
@radimvaculik Alright, like that?

**Comment by radimvaculik** [comment]
Good to me. But I have zero experience with Doctrine. Can anyone else have a look to check the PR?

**Comment by juniwalk** [comment]
@radimvaculik This does not affect Doctrine per se. It just changes the way DataGrind handles Enums returned from Doctrine. Without this DataGrid would not render Status row as it was unable to handle BackedEnum instance.

**Comment by juniwalk** [comment]
Actually if you had for example ArraySource with Enum instance it would see the same issue. DataGrid cannot handle them, mainly in StatusColumn.

**Comment by radimvaculik** [comment]
Good to me.

**Comment by juniwalk** [comment]
@radimvaculik I am thinking if I should add this for other sources as well, besides ArraySource I don't know which one would be able to return instance of Enum.

**Pull request merged**